### ISSUE TITLE
Hot reloading and atom state not found error with React Native expo app

### ISSUE THREAD

**Issue by TehNrd** [pmndrs/jotai#216]
I don't necessarily think this needs to be fixed but I wanted to log this issue should someone else comes across the same problem.

It would appear the hot loading feature of a React Native app with Expo breaks this library if the atom is defined in the same file/component you are editing.

Steps to reproduce.

Create a component that uses an atom, the atom should be defined in the same file.

```
const UserState = atom<null | {}>(null);

export default function useAuth() {
  const [user, setUser] = useAtom(UserState);
 ... //do something with setUser();
}
```

Edit and save the file, the component will hot relaod. Try to call `setUser()` again. You will get an error like so...

```
[Bug] Atom state not found. Please file an issue with repro: delAtom, Set {
  Object {
    "init": null,
    "key": 2,
    "read": [Function anonymous],
    "write": [Function anonymous],
  },
}
- node_modules/react-native/Libraries/LogBox/LogBox.js:117:10 in registerWarning
- node_modules/react-native/Libraries/LogBox/LogBox.js:63:8 in warnImpl
- node_modules/react-native/Libraries/LogBox/LogBox.js:36:4 in console.warn
- node_modules/expo/build/environment/react-native-logs.fx.js:18:4 in warn
- node_modules/jotai/index.cjs.js:180:4 in <global>
- node_modules/jotai/index.cjs.js:499:31 in del
- node_modules/jotai/index.cjs.js:496:10 in del
* [native code]:null in forEach
- node_modules/jotai/index.cjs.js:495:27 in del
* http://127.0.0.1:19001/node_modules/expo/AppEntry.bundle?platform=ios&dev=true&hot=false&minify=false:158971:9 in <unknown>
- node_modules/jotai/index.cjs.js:649:20 in runWriteThunk
- node_modules/jotai/index.cjs.js:769:14 in writeAtom$argument_5
- node_modules/jotai/index.cjs.js:606:2 in writeAtom
- node_modules/regenerator-runtime/runtime.js:63:36 in tryCatch
- node_modules/regenerator-runtime/runtime.js:293:29 in invoke
- node_modules/regenerator-runtime/runtime.js:63:36 in tryCatch
- node_modules/regenerator-runtime/runtime.js:154:27 in invoke
- node_modules/regenerator-runtime/runtime.js:164:18 in PromiseImpl.resolve.then$argument_0
- node_modules/react-native/node_modules/promise/setimmediate/core.js:37:13 in tryCallOne
- node_modules/react-native/node_modules/promise/setimmediate/core.js:123:24 in setImmediate$argument_0
- node_modules/react-native/Libraries/Core/Timers/JSTimers.js:130:14 in _callTimer
- node_modules/react-native/Libraries/Core/Timers/JSTimers.js:181:14 in _callImmediatesPass
- node_modules/react-native/Libraries/Core/Timers/JSTimers.js:441:30 in callImmediates
- node_modules/react-native/Libraries/BatchedBridge/MessageQueue.js:387:6 in __callImmediates
- node_modules/react-native/Libraries/BatchedBridge/MessageQueue.js:135:6 in __guard$argument_0
- node_modules/react-native/Libraries/BatchedBridge/MessageQueue.js:364:10 in __guard
- node_modules/react-native/Libraries/BatchedBridge/MessageQueue.js:134:4 in flushedQueue
* [native code]:null in flushedQueue
* [native code]:null in invokeCallbackAndReturnFlushedQueue
```

I'm guessing this is because the hot reload rebuilds the file and destroys any reference to the atom. 

The fix was pretty easy...after way too much time debugging, just move the atom to a different file that isn't rebuilt on save and it will work...

 ../states: (stores, atoms, whatever, naming is hard!)
```
import { atom } from 'jotai';

export const UserState = atom<null | {}>(null);
```

```
import { UserState } from '../states'; // do this

export default function useAuth() {
  const [user, setUser] = useAtom(UserState);
 ... //do something with setUser();
}
```

**Comment by dai-shi** [comment]
Thanks for reporting.
And, yes it's expected because atom object reference could be changed on hot reloading.
We might be able to improve the message as it's not really a bug.

To confirm it's expected and possibly improve the message, it would be very appreciated if someone created a minimal codesandbox repro with react-dom. Help wanted.

**Comment by dai-shi** [comment]
Closing this. Please open a new issue/discussion with a repro if any.
The real fix would be #487.
